### PR TITLE
[TA] Abstractive Summarization and Extractive Summarization beSpokeMethod

### PR DESCRIPTION
**Pull request by mssfang**
# Description

- Added bespoke methods for the abstractive summarization and extractive summarization,
    - `beginAbstractSummary(...)`
    - `beginExtractSummary(...)`
- Added new models for the new bespoke methods
    - `AbstractSummaryOperationDetail`, `AbstractSummaryOptions`, `AbstractSummaryPagedFlux`, `AbstractSummaryPagedIterable`
    - `ExtractSummaryOperationDetail`, `ExtractSummaryOptions`, `ExtractSummaryPagedFlux`, `ExtractSummaryPagedIterable`

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,12 @@
 ## 5.3.0-beta.2 (Unreleased)
 
 ### Features Added
+- Added bespoke methods for the abstractive summarization and extractive summarization,
+    - `beginAbstractSummary(...)`
+    - `beginExtractSummary(...)`
+- Added new models for the new bespoke methods
+    - `AbstractSummaryOperationDetail`, `AbstractSummaryOptions`, `AbstractSummaryPagedFlux`, `AbstractSummaryPagedIterable`
+    - `ExtractSummaryOperationDetail`, `ExtractSummaryOptions`, `ExtractSummaryPagedFlux`, `ExtractSummaryPagedIterable`
```
Comment (deyaaeldeen) [inline sdk/textanalytics/azure-ai-textanalytics/CHANGELOG.md:11]: nitpick: wording:
```suggestion
- Added the following methods for performing abstractive summarization and extractive summarization actions:
    - `beginAbstractSummary(...)`
    - `beginExtractSummary(...)`
- Added the following types for the newly added methods
    - `AbstractSummaryOperationDetail`, `AbstractSummaryOptions`, `AbstractSummaryPagedFlux`, `AbstractSummaryPagedIterable`
    - `ExtractSummaryOperationDetail`, `ExtractSummaryOptions`, `ExtractSummaryPagedFlux`, `ExtractSummaryPagedIterable`
```

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/ExtractSummaryUtilClient.java [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+import static com.azure.ai.textanalytics.implementation.Utility.toMultiLanguageInput;
+import static com.azure.ai.textanalytics.implementation.models.State.CANCELLED;
+import static com.azure.ai.textanalytics.implementation.models.State.NOT_STARTED;
+import static com.azure.ai.textanalytics.implementation.models.State.RUNNING;
+import static com.azure.ai.textanalytics.implementation.models.State.SUCCEEDED;
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.tracing.Tracer.AZ_TRACING_NAMESPACE_KEY;
+
+/**
+ * Helper class for managing abstractive summarization endpoints.
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/ExtractSummaryUtilClient.java:82]:
```suggestion
 * Helper class for managing extractive summarization endpoints.
```

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java [hunk 3]
```diff
@@ -2692,6 +2700,372 @@ public Response<DynamicClassifyDocumentResultCollection> dynamicClassificationBa
...
+     *         documentsResults.getProjectName&#40;&#41;, documentsResults.getDeploymentName&#40;&#41;&#41;;
+     *     for &#40;ClassifyDocumentResult documentResult : documentsResults&#41; &#123;
+     *         System.out.println&#40;&quot;Document ID: &quot; + documentResult.getId&#40;&#41;&#41;;
+     *         for &#40;ClassificationCategory classification : documentResult.getClassifications&#40;&#41;&#41; &#123;
+     *             System.out.printf&#40;&quot;&#92;tCategory: %s, confidence score: %f.%n&quot;,
+     *                 classification.getCategory&#40;&#41;, classification.getConfidenceScore&#40;&#41;&#41;;
+     *         &#125;
+     *     &#125;
+     * &#125;&#41;;
+     * </pre>
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java:2736]: seems like the wrong sample (edits: same for all the variants of this method)

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/ExtractSummaryOptions.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    public String getDisplayName() {
+        return displayName;
+    }
+
+    /**
+     * Sets display name of the operation.
+     *
+     * @param displayName Display name of the operation.
+     *
+     * @return The {@link AbstractSummaryOptions} object itself.
```
Comment (kristapratico) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/ExtractSummaryOptions.java:32]:
```suggestion
     * @return The {@link ExtractSummaryOptions} object itself.
```

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java (hunk 3 above)
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClient.java:2736]: I forget to update the codesnap src_embed names. Thanks Krista.

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AbstractSummaryUtilClient.java [hunk 5]
```diff
@@ -0,0 +1,422 @@
...
+                pollingOperationSync(operationId -> service.jobStatusWithResponse(operationId,
+                    finalIncludeStatistics, null, null, finalContext)),
+                cancelOperationSync(operationId -> service.cancelJobWithResponse(operationId, finalContext)),
+                fetchingOperationIterable(
+                    operationId -> getAbstractSummaryPagedIterable(operationId, null, null,
+                        finalIncludeStatistics, finalContext)));
+
+
+        } catch (ErrorResponseException ex) {
+            throw LOGGER.logExceptionAsError((HttpResponseException) mapToHttpResponseExceptionIfExists(ex));
```
[the remark this answers is not in the document]
Comment (mssfang) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/AbstractSummaryUtilClient.java:172]: Good point. ErrorResponseException extends HttpResponseException. It should be doable.

**Comment by mssfang** [comment]
will create another PR for renaming the method and class names based on the latest discussion within v-teams

**Review by kristapratico — APPROVED** [review]

**Pull request merged**